domain/resource/service/create.go

```diff
@@ -0,0 +1,65 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"miniWiki/domain/resource/model"
+	"miniWiki/domain/resource/query"
+	"miniWiki/utils"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/sirupsen/logrus"
+)
+
+func (s *Resource) CreateResource(ctx context.Context, request model.CreateResourceRequest) (*model.ResourceResponse, error) {
+	err := s.validateCategoryOwner(ctx, request.Resource.CategoryId, request.AccountId)
+	if err != nil {
+		return nil, err
+	}
+
+	params := query.InsertResourceParams{
+		Title:       request.Resource.Title,
+		Description: request.Resource.Description,
+		Link:        request.Resource.Link,
+		CategoryID:  request.Resource.CategoryId,
+		AuthorID:    request.AccountId,
+		State:       query.ResourceState(strings.ToUpper(request.Resource.State)),
+	}
+
+	res, err := s.resourceQuerier.InsertResource(ctx, params)
+	if err != nil {
+		logrus.WithContext(ctx).
+			Errorf("Failed inserting in database: %v", err)
+	}
+
+	return &model.ResourceResponse{
+		ResourceId: res,
+	}, err
+}
+
+func (s *Resource) validateCategoryOwner(ctx context.Context, categoryId int, accountId int) error {
```

Comment:
Why are categories owned by specific accounts?

Reply:
Well, if we don't, then any account can modify/delete categories or they shouldn't be modifiable/deletable at all? If we let anyone use the category but not to edit/remove then what happens when the owner does it ?

Reply:
Deleted category validation in new update